statsmodels/genmod/generalized_estimating_equations.py

```diff
@@ -461,6 +461,19 @@ def unpack_cov(self, bcov):
 
 
 def _check_args(endog, exog, groups, time, offset, exposure):
+    # deal with lists and tuples up-front
```

Comment:
You should just use `asarray` on all of them without a check. This function doesn't return anything so it is safe to convert.  Note that asarray won't covert existing NumPy arrays, but will convert anything that is list like (including other sequences other than list or tuple) to ndarray, which is the best approach.